lib/instances.js

```diff
@@ -157,10 +158,15 @@ exports.getControllerKey = async ({id}) => {
 
 // set issuer and capabilities (zcaps) for an instance
 exports.setIssuer = async ({id, controller, issuer, zcaps}) => {
+  assert.string(id, 'id');
   const query = {'instance.id': id};
   if(controller) {
+    if(!Array.isArray(controller)) {
```

Comment:
Was this done to handle the case for when `controller` is an Array? I don't think the API should support multiple controllers at once trying to set properties on the Issuer Instance. Why not assert the controller property to be a string.

Reply:
@dlongley this one is for you.

Reply:
if controller is an array in the database, you don't use `$in`, doing `controller: controller` just works.  Is that what is wanted here?

Reply:
@mattcollier That should work.

https://docs.mongodb.com/manual/tutorial/query-arrays/#query-an-array-for-an-element

Reply:
New implementation now.